src/ConciseCasts.h

```diff
@@ -55,42 +55,71 @@ inline Expr u8(Expr e) {
     return cast(UInt(8, e.type().lanes()), e);
 }
 
+namespace Internal {
+
+inline Expr signed_saturating_cast(Expr e, int bits) {
+    if (e.type().element_of() == Int(bits)) {
+        return e;
+    }
+    if (bits <= e.type().bits()) {
```

Comment:
Actually, I just saw that we do have 64 bit saturating casts, so this needs to be handled.